robot/Cumulus/tests/browser/gift_entry/test_batch_status_adding_removing_gifts.robot

```diff
@@ -0,0 +1,141 @@
+    ...                              Adds the status field. Initiates Batch and adds different
+    ...                              Gift entries. Validates the different Status messages.
+
+    [tags]                           unstable               W-043100            feature:GE
+
+    Edit Default Template To Add Batch Table Columns
+    Go To Page                       Landing                       GE_Gift_Entry
+    Click Gift Entry Button          New Batch
+    Wait Until Modal Is Open
+    Select Template                  Default Gift Entry Template
+    Current Page Should Be           Form                          GE_Gift_Entry
+    Fill Gift Entry Form
+    ...                              Batch Name=${MSG}
+    ...                              Batch Description=This is a test batch created via automation script
+    Click Gift Entry Button          Next
+    Click Gift Entry Button          Save
+    Current Page Should Be           Form                          Gift Entry
+
+    ${BATCH_ID} =                    Save Current Record ID For Deletion     ${NS}DataImportBatch__c
+
+    Fill Gift Entry Form
+    ...                              Donor Type=Contact1
+    ...                              Existing Donor Contact=${CONTACT}[Name]
+    ...                              Donation Amount=${AMOUNT}
+    ...                              Donation Date=Today
+
+    Click Special Button             Save & Enter New Gift
+    Verify Table Field Values        Batch Gifts
+    ...                              Status=Dry Run - Validated
+
+    # Update the gift value and check the value gets  reflected in the table
+    Perform Action On Datatable Row   ${CONTACT}[Name]          Open
+    Fill Gift Entry Form
+     ...                              Donation Amount=${AMOUNT_UPDATED}
+    Click Special Button              Update
+    Verify Table Field Values         Batch Gifts
+     ...                              Donation Amount=$${AMOUNT_UPDATED}.00
+    # Run and validate the Batch data import status
+
+    Process And Validate Batch       Completed
+    Verify Table Field Values        Batch Gifts
+    ...                              Status=Imported
+    # Verify the backend dataimport batch job using the API call
+    Verify Expected Values    nonns  DataImportBatch__c    ${BATCH_ID}
+    ...                              Batch_Status__c=Completed
+
+    #Scroll to the top of the page, add gift for contact associated with org account
+    Scroll Page To Location          0      0
+    Fill Gift Entry Form
+    ...                              Donor Type=Contact1
+    ...                              Existing Donor Contact=${CONTACT2}[Name]
+    ...                              Donation Amount=${AMOUNT}
+    ...                              Donation Date=Today
+
+    Click Special Button             Save & Enter New Gift
+    Verify Table Field Values        Batch Gifts
```

Comment:
at this point, u have 2 gifts in the table. How r u verifying the correct row has status of Dry Run - Error?

Reply:
I modified the existing keyword to support this functionality.